cdap-watchdog/src/main/java/co/cask/cdap/logging/write/LogCleanup.java

```diff
@@ -67,14 +71,18 @@ public void handle(Location location) {
                                                               location.toURI()), e);
                                               throw Throwables.propagate(e);
                                             }
+
+                                            // Delete any empty parent dirs
+                                            for (Location dir : parentDirs) {
```

Comment:
how come this was moved into the callback? This runs in a transaction right?  If it should stay here we can move parentDirs into here instead of making it final outside.